plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/server/LocalAppEngineServerLaunchConfigurationDelegate.java

```diff
@@ -369,7 +371,10 @@ public void launch(ILaunchConfiguration configuration, String mode, final ILaunc
         int debugPort = getDebugPort();
         setupDebugTarget(devServerRunConfiguration, launch, debugPort, monitor);
       }
-      IVMInstall vmInstall = getVMInstall(configuration);
+      
+      IJavaProject javaProject = JavaCore.create(modules[0].getProject());
```

Comment:
Also not clear how to compare different `IVMInstall`s to select the best one.  THere's some stuff we can do with the `IExecutionEnvironment`s (an IEE can be obtained from a project by examining its associated JRE, as per `ComplianceConfigurationBlock#getEE()`).

But it might be best to just integrate the _Java Runtime Block_ into a launch config tab, and then `getVMInstall(ILaunchConfiguration)` may do the right thing.